libchisel/src/remapimports.rs

```diff
@@ -1,242 +1,92 @@
 use std::collections::HashMap;
 
-use super::{ModuleError, ModulePreset, ModuleTranslator};
+use super::{
+    imports::{ImportList, ImportType},
+    ModuleError, ModulePreset, ModuleTranslator,
+};
 use parity_wasm::elements::*;
 
-#[derive(Hash, Eq, PartialEq, Debug, Clone)]
-pub struct ImportPair {
-    pub module: String,
-    pub field: String,
+pub struct RemapImports<'a> {
+    /// The list of correct imports.
+    list: ImportList<'a>,
+    /// The prefix that compiler-generated import names contain. In the case of ewasm, this is
```

Comment:
I think this is a bad direction. This module supposed to be very generic accepting a mapping from one import to another. The correct way to solve here if we don't want to have a big manual list is to have a helper which generates this prefixed version out of the official "ewasm" import list.

Reply:
Agreed. Although while this works for ewasm, I think it is wise to update this for 0.3.0 instead.

Reply:
@axic #70

Reply:
> Although while this works for ewasm, I think it is wise to update this for 0.3.0 instead.

Not sure what you mean. Do mean to first remove the functionality we have and then add it back?

Reply:
I mean that this works for the current use cases. We can make the prefix feature optional or remove it later.

Reply:
For eth2 this is now badly needed.